### PR TITLE
Easier way to disable events

### PR DESCRIPTION
**Pull request by abhijoshi2k**
Thank you for contributing! Please check the following things before submiting your PR:

Required:
- [x] I have read and followed the [contributing guidelines](https://github.com/Readme-Workflows/recent-activity/blob/main/.github/CONTRIBUTING.md)

If necessary:
- [ ] I have updated the README and documentation.
- [ ] I have updated the ChangeLog with the changes I have made.

Added easier way to disable events.
Following inputs will no longer be accepted:
```
DISABLE_COMMENTS
DISABLE_PR
DISABLE_ISSUES
```

It will be replaced by single input: `DISABLE_EVENTS`.
This will be a string containing comma separated names of events to be disabled.

Example:
`DISABLE_EVENTS: 'comments, issues'`

---------------------------------------------------------------------

**Note:** we will close your PR without comment if you do not check the required boxes above and provide ALL requested information.

### CODE REVIEW

**Comment by abhijoshi2k** [comment]
@Andre601 @PuneetGopinath Please test the following and then update changelog and readme on same branch.
We will release after that.

**Comment by Andre601** [comment]
Isn't there any way for it to provide a String list or String Array?

**Comment by abhijoshi2k** [comment]
> Isn't there any way for it to provide a String list or String Array?

In that case, the input will have to be a stringified array which will again contain strings.
f.e. `DISABLE_EVENTS: '["comments", "issues"]'`

It is a complex syntax I feel.
And if user fails to put it in this syntax, workflow will fail!
So I went for string.

**Comment by Andre601** [comment]
> > Isn't there any way for it to provide a String list or String Array?
> 
> In that case, the input will have to be a stringified array which will again contain strings.
> f.e. `DISABLE_EVENTS: '["comments", "issues"]'` (single quotes included)
> 
> It is a complex syntax I feel.
> And if user fails to put it in this syntax, workflow will fail!
> So I went for string.

I assumed GitHub usually does some transformations or similar, since you often can put both a String Array or String List and it would be handled or something.

Also, I really believe we should stop thinking that people are *that* stupid.

File: dist/index.js [hunk 1]
```diff
@@ -15250,11 +15250,11 @@ const ISSUE_CLOSED = core.getInput("ISSUE_CLOSED");
 const PR_OPENED = core.getInput("PR_OPENED");
 const PR_CLOSED = core.getInput("PR_CLOSED");
 const PR_MERGED = core.getInput("PR_MERGED");
-const DISABLE_COMMENTS = core.getInput("DISABLE_COMMENTS");
-const DISABLE_ISSUES = core.getInput("DISABLE_ISSUES");
-const DISABLE_PR = core.getInput("DISABLE_PR");
 const URL_TEXT = core.getInput("URL_TEXT");
 
+let DISABLE_EVENTS = core.getInput("DISABLE_EVENTS").toLowerCase().split(",");
```
Comment (Andre601) [inline dist/index.js:15255]: I assume there won't be any issues regarding spaces or similar? Like when I f.e. write `"COMMENTS, ISSUES ,PR"` that it won't fail due to formatting.

Comment (abhijoshi2k) [reply dist/index.js:15255]: No there won't be any such issues as I change it to lowercase and trim it to remove extra spaces.

**Review by Andre601 — APPROVED** [review]

**Comment by abhijoshi2k** [comment]
> What are the events for the String?
> So that I can prepare a PR to update the readme (I prefer making a separate PR for readme changes)

The events are `issues`, `pr` and `comments`
Yeah make separate pr. That's okay.

**Review by PuneetGopinath — APPROVED** [review]

**Comment by abhijoshi2k** [comment]
@PuneetGopinath Can you update the changelog please?

**Comment by Andre601** [comment]
So, will this be merged?

**Comment by abhijoshi2k** [comment]
> So, will this be merged?
> (Should I update the changelog?)

Yes this will be merged once changelog and readme are updated.
You can open pr with the branch you are working on (to edit readme) on this branch right?

So we'll merge your fork with this branch. And then merge this branch into main.

And i think you can update changelog too as @PuneetGopinath seems occupied currently.

Then we'll release.

**Comment by Andre601** [comment]
I honestly dislike how the ChangeLog file currently looks like.

Would you guys mind if I change its design and also update the name to CHANGELOG to have it more consistent?

**Comment by abhijoshi2k** [comment]
> I honestly dislike how the ChangeLog file currently looks like.
> 
> Would you guys mind if I change its design and also update the name to CHANGELOG to have it more consistent?

Yes that'd be fine.

**Comment by abhijoshi2k** [comment]
> Something else related to changelogs...
> 
> Perhaps have a GitHub Action to automate the creation of the changelog file?
> Downside would probably be that it would update AFTER the release, but since people are supposed to always use the latest release should this be a minor downside.

I'll look for it on marketplace

**Pull request merged**